framework/src/preconditioners/PhysicsBasedPreconditioner.C

```diff
@@ -113,16 +113,20 @@ PhysicsBasedPreconditioner::PhysicsBasedPreconditioner(const InputParameters & p
   unsigned int n_vars = _nl.system().n_vars();
 
   // off-diagonal entries
-  const std::vector<NonlinearVariableName> & odr =
-      getParam<std::vector<NonlinearVariableName>>("off_diag_row");
-  const std::vector<NonlinearVariableName> & odc =
-      getParam<std::vector<NonlinearVariableName>>("off_diag_column");
   std::vector<std::vector<unsigned int>> off_diag(n_vars);
-  for (const auto i : index_range(odr))
+  if (isParamValid("off_diag_row") && isParamValid("off_diag_column"))
   {
-    unsigned int row = _nl.system().variable_number(odr[i]);
-    unsigned int column = _nl.system().variable_number(odc[i]);
-    off_diag[row].push_back(column);
+    const std::vector<NonlinearVariableName> & odr =
+        getParam<std::vector<NonlinearVariableName>>("off_diag_row");
+    const std::vector<NonlinearVariableName> & odc =
+        getParam<std::vector<NonlinearVariableName>>("off_diag_column");
+
+    for (const auto i : index_range(odr))
+    {
+      unsigned int row = _nl.system().variable_number(odr[i]);
+      unsigned int column = _nl.system().variable_number(odc[i]);
+      off_diag[row].push_back(column);
+    }
   }
   // Add all of the preconditioning systems
```

Comment:
I think you can move this inside the isParamValid condition

Reply:
It's already been inside the isParamValid condition. That's what the line 130 `}` for

Reply:
131 should stay with `addSystem(var, off_diag[var], _pre_type[var]);` Put `AddSystem` inside will cause three test cases to fail.
`problems/eigen_problem/preconditioners.newton_pbp ........................... FAILED (EXPECTED OUTPUT MISSING)
restart/restart_add_variable.add_variable_restart ............................................. FAILED (CRASH)
preconditioners/pbp.lots_of_variables ......................................................... FAILED (CRASH)
`